pkg/common/cns-lib/volume/manager.go

```diff
@@ -565,63 +576,184 @@ func (m *defaultManager) DeleteVolume(ctx context.Context, volumeID string, dele
 			log.Errorf("ConnectCns failed with err: %+v", err)
 			return err
 		}
+		if m.idempotencyHandlingEnabled {
+			return m.deleteVolumeWithIdempotency(ctx, volumeID, deleteDisk)
+		}
+		return m.deleteVolumeWithoutIdempotency(ctx, volumeID, deleteDisk)
+	}
+	start := time.Now()
+	err := internalDeleteVolume()
+	if err != nil {
+		prometheus.CnsControlOpsHistVec.WithLabelValues(prometheus.PrometheusCnsDeleteVolumeOpType,
+			prometheus.PrometheusFailStatus).Observe(time.Since(start).Seconds())
+	} else {
+		prometheus.CnsControlOpsHistVec.WithLabelValues(prometheus.PrometheusCnsDeleteVolumeOpType,
+			prometheus.PrometheusPassStatus).Observe(time.Since(start).Seconds())
+	}
+	return err
+}
+
+// deleteVolumeWithoutIdempotency attempts to delete the volume on CNS. CNS task information is not persisted and callers
+// cannot assume that operations are idempotent.
+func (m *defaultManager) deleteVolumeWithoutIdempotency(ctx context.Context, volumeID string, deleteDisk bool) error {
```

Comment:
Can you confirm that there is no functional change here?

Reply:
Confirmed, No functional changes